custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -333,10 +333,17 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
 
     vm.onSelectYear = function (item) {
         if (item.id === new Date().getFullYear()) {
-            vm.months = _.filter(vm.monthsCopy, function(month) {
-                return month.id <= new Date().getMonth() + 1;
-            });
-            vm.selectedMonth = vm.selectedMonth <= new Date().getMonth() + 1 ? vm.selectedMonth : new Date().getMonth() + 1;
+            if (vm.isIncentiveReportSelected() && new Date().getDate() < 15) {
```

Comment:
Does this code do the following:

If its before the 15th, show months before the previous month. If its the 15th or later, show all months including the previous month?

Reply:
Close, except replace previous month with current month

Reply:
Ah, i now realize that it should do what you are describing, although does what i am describing. Will fix